app/src/main/java/com/google/android/stardroid/activities/EditSettingsActivity.java

```diff
@@ -84,15 +84,32 @@ public void onCreate(Bundle savedInstanceState) {
   public void onStart() {
     super.onStart();
     analytics.trackPageView(Analytics.EDIT_SETTINGS_ACTIVITY);
-    Preference locationPreference = preferenceFragment.findPreference(LOCATION);
+    final Preference locationPreference = preferenceFragment.findPreference(LOCATION);
+    Preference latitudePreference = preferenceFragment.findPreference(LATITUDE);
+    final Preference longitudePreference = preferenceFragment.findPreference(LONGITUDE);
     // TODO(johntaylor) if the lat long prefs get changed manually, we should really
     // reset the placename to "" too.
     locationPreference.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
     
       public boolean onPreferenceChange(Preference preference, Object newValue) {
         Log.d(TAG, "Place to be updated to " + newValue);
-        boolean success = setLatLongFromPlace(newValue.toString());
-        return success;
+        return setLatLongFromPlace(newValue.toString());
+      }
+    });
+  
+    latitudePreference.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
+    
+      public boolean onPreferenceChange(Preference preference, Object newValue) {
+        ((EditTextPreference)locationPreference).setText("");
```

Comment:
Minor style nit (here and line 111) can you include a space after the cast before the variable name?  

https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace

Reply:
@jaydeetay Thanks for reviewing! I'll just push the changes.